corpus/data_augmentation.py

```diff
@@ -236,3 +243,167 @@ def run(self):
 
         with open(str(self.out_segment_file), "w") as segments_outfile:
             segments_outfile.writelines(segment_file_names)
+
+
+class AddImpulseResponseJob(Job):
+    """
+    Add impulse responses to each recording in the corpus with a given probability to be affected.
+    This is done by convolving the audio with an impulse response to simulate other acoustic conditions.
+    Supports .wav files
+
+    This job is dependent on the audiomentations library.
+    """
+
+    def __init__(
+        self, corpus_file, new_corpus_name, ir_path, augment_prob, time_rqmt=12
+    ):
+        """
+
+        :param Path corpus_file: Bliss corpus with wav files
+        :param str new_corpus_name: name of the new corpus
+        :param str ir_path: path to the directory containing impulse responses
+        :param float augment_prob: probability of each audio to be augmented by impulse response convolution
+        :param int time_rqmt: time requirement for the Sisypus job
+        """
+        self.corpus_file = corpus_file
+        self.new_corpus_name = new_corpus_name
+        self.impulse_responses_dir = ir_path
+
+        self.ir_path = ir_path
+        self.augment_prob = augment_prob
+
+        self.rqmt = {"time": time_rqmt, "cpu": 2}
+
+        self.out_audio_folder = self.output_path("out_audio/", directory=True)
+        self.out_corpus = self.output_path("ir_convoluted.corpus.xml.gz")
+
+    def tasks(self):
+        yield Task("run", rqmt=self.rqmt)
+
+    def run(self):
+        add_impulse_response = audiomentations.AddImpulseResponse(
+            ir_path=self.ir_path, p=self.augment_prob
+        )
+
+        job_id = os.path.basename(self.job_id())
+        orig_corpus = corpus.Corpus()
+        new_corpus = corpus.Corpus()
```

Comment:
I will test if it is not better to modify the corpus in-place instead of creating a new one, that way also sub-corpus structures would be preserved (which would currently break).

Reply:
>   sub-corpus structures [...] would currently break

They would be flattened but otherwise the functionality would stay the same.
But I agree nothing speaks against preserving the original structure.

Reply:
But if they are flattened the "full" segment names change, and then this could cause issues.

Reply:
Ah, you are right, I agree.